libs/wallet/src/connectors/snap-connector.ts

```diff
@@ -1,247 +1,282 @@
+    // if more events get added we will need more logic here
+    this.pollRef = setInterval(async () => {
+      const result = await this.isConnected();
+      if (result.connected) return;
+      this.emit('client.disconnected');
+    }, 2000);
+  }
 
-    if (!result?.transaction?.signature) {
-      throw new Error('could not retrieve transaction siganture');
+  stopPoll() {
+    if (this.pollRef) {
+      clearInterval(this.pollRef);
     }
+  }
 
-    return {
-      transactionHash: result.transactionHash,
-      signature: result?.transaction?.signature?.value,
-      receivedAt: result.receivedAt,
-      sentAt: result.sentAt,
-    };
+  /**
+   * Requests permission for a website to communicate with the specified snaps
+   * and attempts to install them if they're not already installed.
+   * If the installation of any snap fails, returns the error that caused the failure.
+   * More informations here: https://docs.metamask.io/snaps/reference/rpc-api/#wallet_requestsnaps
+   */
+  async requestSnap() {
+    await this.request(EthereumMethod.RequestSnaps, {
+      [this.snapId]: {
+        version: this.version,
+      },
+    });
   }
 
-  async getChainId(): Promise<GetChainIdResponse> {
-    if (!this.nodeAddress) throw SnapConnectorErrors.NODE_ADDRESS_NOT_SET;
-    if (!this.snapId) throw SnapConnectorErrors.SNAP_ID_NOT_SET;
+  // TODO: check if this is needed, its used in use-snap-status
+  //
+  //
+  // /**
+  //  * Gets the list of all installed snaps.
+  //  * More information here: https://docs.metamask.io/snaps/reference/rpc-api/#wallet_getsnaps
+  //  */
+  // async getSnap() {
+  //   const snaps = await this.request(EthereumMethod.GetSnaps);
+  //   return Object.values(snaps).find(
+  //     (s) => s.id === this.snapId && s.version === this.version
+  //   );
+  // }
 
-    const response = await invokeSnap<GetChainIdResponse>(this.snapId, {
-      method: 'client.get_chain_id',
-      params: { networkEndpoints: [this.nodeAddress] },
+  /**
+   * Calls a method on the specified snap, always vega in this case
+   * should always be npm:@vegaprotocol/snap
+   */
+  async invokeSnap<TResult>(
+    method: JsonRpcMethod,
+    params?: SnapInvocationParams
+  ): Promise<TResult> {
+    return await this.request(EthereumMethod.InvokeSnap, {
+      snapId: this.snapId,
+      request: {
+        method,
+        params,
+      },
     });
-
-    return response;
   }
 
-  async disconnect() {
-    clearConfig();
+  /**
+   * Calls window.ethereum.request with method and params
+   */
+  async request<TResult>(
+    method: EthereumMethod,
+    params?: object
+  ): Promise<TResult> {
+    if (
+      'ethereum' in window &&
+      typeof window.ethereum === 'object' &&
+      window.ethereum &&
+      'request' in window.ethereum &&
+      'isMetaMask' in window.ethereum &&
+      window.ethereum.isMetaMask &&
+      typeof window.ethereum.request === 'function'
```

Comment:
Can this not just be done with simple optional chaining?

```suggestion
      window.ethereum?.isMetaMask &&
      typeof window.ethereum?.request === 'function'
```